### PR TITLE
Fix bug with mkl impl of spgemm

### PR DESCRIPTION
**Pull request by ndellingwood**

### CODE REVIEW

**Comment by ndellingwood** [comment]
Matching fix to trilinos/Trilinos#9891.
This is only needed for master branch, was resolved on develop branch by PR #1135 with the clang-format enhancement which was not included in 3.5.0 release

File: src/sparse/impl/KokkosSparse_spgemm_mkl_impl.hpp [hunk 1]
```diff
@@ -425,12 +424,6 @@ void mkl_symbolic(
      MKL\n"); return;
       }
   */
-  if (std::is_same<idx, int>::value) {
```
Comment (e10harvey) [inline src/sparse/impl/KokkosSparse_spgemm_mkl_impl.hpp:428]: I don't see what this if-block ends; is there a matching `}` for it? Should the if-block just be converted to a block -- `{`.

**Review by e10harvey — CHANGES_REQUESTED** [review]
THanks, @ndellingwood ! Please see questions below.

File: src/sparse/impl/KokkosSparse_spgemm_mkl_impl.hpp (hunk 1 above)
Comment (ndellingwood) [reply src/sparse/impl/KokkosSparse_spgemm_mkl_impl.hpp:428]: @e10harvey there is not a matching brace to close this if block, that was what was causing the compilation error. There was an apparent copy+paste error in a past update to this file that duplicated the lines of code that I removed in this PR (and so removing the mismatch in open-closed braces)

Comment (ndellingwood) [reply src/sparse/impl/KokkosSparse_spgemm_mkl_impl.hpp:428]: @e10harvey See lines 433-448 of this file which had mistaken duplication to lines 428-431, removed by this PR

**Review by e10harvey — APPROVED** [review]

**Comment by e10harvey** [comment]
@ndellingwood, @lucbv: Shall I enable the auto-tester for pull requests targeting `master`?

**Comment by ndellingwood** [comment]
Alright, looks like the issue was never on the develop branch but had to do with problems with a cherry-pick to the release-candidate. The duplication occurred with commit 05986906f , here is the diff https://github.com/kokkos/kokkos-kernels/pull/1149/commits/05986906f4acce3d2091f6181db2b099a6780945

**Pull request merged**